ui/src/components/crop_image_tool/mod.rs

```diff
@@ -0,0 +1,155 @@
+                onclick: move |_| {},
+                div {
+                    id: "crop-image-topbar", 
+                    background: "var(--secondary)",
+                    height: "70px",
+                    border_radius: "12px",
+                    div {
+                        id: "crop-image-topbar-left",
+                        padding: "16px",
+                        display: "inline-flex",
+                        div {
+                            id: "crop-image-topbar-left-title",
+                            color: "var(--text-color)",
+                            margin_right: "32px",
+                            "Please select the area\n you want to crop"
+                        },
+                        Button {
+                            appearance: Appearance::DangerAlternative,
+                            icon: Shape::XMark,
+                            onpress: move |_| {
+                                cx.props.on_cancel.call(());
+                                crop_image.set(false);
+                            }
+                        },
+                        div {
+                            margin_right: "16px",
+                        }
+                        Button {
+                            appearance: Appearance::Success,
+                            icon: Shape::Check,
+                            onpress: move |_| {
+                                cx.spawn({
+                                    to_owned![eval, image_scale, cropped_image_pathbuf, clicked_button_to_crop];
+                                    async move {
+                                        let save_image_cropped_js = include_str!("./save_cropped_image.js")
+                                        .replace("$IMAGE_SCALE", (1.0 / *image_scale.read()).to_string().as_str());
+                                        if let Ok(r) = eval(&save_image_cropped_js) {
+                                            if let Ok(val) = r.join().await {
+                                                let thumbnail = val.as_str().unwrap_or_default();
+                                                let base64_string = thumbnail.trim_matches('\"');
+                                                let decoded_bytes = match base64::decode(base64_string) {
+                                                    Ok(bytes) => bytes,
+                                                    Err(e) => {
+                                                        log::error!("Error decoding base64 string for cropped image: {}", e);
+                                                        return;
+                                                    },
+                                                };
+                                                let cropped_image_path = STATIC_ARGS.uplink_path.join("cropped_image.png");
+                                                let mut file = File::create(cropped_image_path.clone()).unwrap();
```

Comment:
Since we are within a async block, we should use `tokio::fs::File` instead and await its functions since `File` and its IO in std is blocking.